### PR TITLE
Fix CursorEntered event crash

### PR DESCRIPTION
**Pull request by j4qfrost**
This change is dependent on https://github.com/dvc94ch/winit/pull/1

After tinkering with the app with the fix, I discovered an issue with holding down a mouse button. I will investigate further.

### CODE REVIEW

File: flutter-winit/src/window.rs [hunk 1]
```diff
@@ -184,6 +186,7 @@ impl FlutterWindow {
             localization.send_locale(locale_config::Locale::current());
         });
 
+        let mut active_devices: HashMap<i32, DeviceId> = HashMap::new();
```
Comment (csnewman) [inline flutter-winit/src/window.rs]: Can we move `active_devices` inside `FlutterEngine`? As we know its will always cause a crash, we may as well give everyone the sanity check

File: flutter-winit/src/window.rs [hunk 2]
```diff
@@ -192,31 +195,42 @@ impl FlutterWindow {
...
-                                FlutterPointerPhase::Add,
-                                cursor,
-                                FlutterPointerSignalKind::None,
-                                (0.0, 0.0),
-                                FlutterPointerDeviceKind::Mouse,
-                                FlutterPointerMouseButtons::Primary,
-                            );
+                        WindowEvent::CursorEntered { device_id, .. } => {
+                            let idx = device_id.get_index();
+                            eprintln!("{:?}", active_devices);
```
Comment (csnewman) [inline flutter-winit/src/window.rs]: I assume this is left over from debugging?

File: flutter-winit/src/window.rs (hunk 1 above)
Comment (csnewman) [reply flutter-winit/src/window.rs]: We can also switch to a `hashset` when we move, as we only care about the index, not the actual device id.

File: flutter-winit/src/window.rs [hunk 3]
```diff
@@ -192,31 +195,42 @@ impl FlutterWindow {
...
+                            match active_devices.get(&idx) {
```
Comment (csnewman) [inline flutter-winit/src/window.rs]: Once moved to `HashSet`, we can use just a single `insert` statement, and use the `bool` resort to see if the value already existed.

File: flutter-winit/src/window.rs (hunk 1 above)
Comment (j4qfrost) [reply flutter-winit/src/window.rs]: I'll see if I can switch it over to FlutterEngine. I'll switch over to hashset.

File: flutter-winit/src/window.rs (hunk 2 above)
Comment (j4qfrost) [reply flutter-winit/src/window.rs]: D'oh!

File: flutter-winit/src/window.rs (hunk 1 above)
Comment (j4qfrost) [reply flutter-winit/src/window.rs]: Actually the crash happens on interaction with the window. We could have each window manage its own devices.

Comment (csnewman) [reply flutter-winit/src/window.rs]: My thoughts are more of, no matter how you're using `FlutterEngine`, sending `send_pointer_event` for a non existent device, or creating two devices will lead to a crash. Why make each window provider duplicate that logic? We can also expose a `is_device_active(i32)` to allow windows to check if a device is active, if needed

**Comment by csnewman** [comment]
@dvc94ch Not sure if you want @j4qfrost to open https://github.com/dvc94ch/winit/pull/1 on https://github.com/rust-windowing/winit instead?

**Comment by dvc94ch** [comment]
Yes that would be better.

**Comment by dvc94ch** [comment]
So the flutter-rs code looks good to me. The winit code needs to be upstreamed and I don't think it'll get accepted in it's current state. I'll leave some comments.

**Comment by dvc94ch** [comment]
I changed my mind. We should just use a `Vec` instead of a `HashSet`. For three device ids or whatever the cost of hashing the device id is probably higher than a linear search. Then we also don't require any changes to winit.

File: flutter-winit/Cargo.toml [hunk 4]
```diff
@@ -15,7 +15,8 @@ flutter-engine-sys = { path = "../flutter-engine-sys" }
 flutter-engine = { path = "../flutter-engine", features = ["texture-registry"] }
 flutter-plugins = { path = "../flutter-plugins" }
 futures-task = "0.3.1"
-glutin = { git = "https://github.com/dvc94ch/glutin", branch = "android" }
+glutin = { path = "../../glutin" }
```
Comment (dvc94ch) [inline flutter-winit/Cargo.toml]: revert

File: flutter-winit/src/window.rs [hunk 5]
```diff
@@ -184,6 +186,7 @@ impl FlutterWindow {
             localization.send_locale(locale_config::Locale::current());
         });
 
+        let mut active_devices: HashSet<i32> = HashSet::new();
```
Comment (dvc94ch) [inline flutter-winit/src/window.rs]: use a `Vec<DeviceId>`

File: flutter-winit/src/window.rs [hunk 6]
```diff
@@ -192,31 +195,39 @@ impl FlutterWindow {
...
-                                FlutterPointerDeviceKind::Mouse,
-                                FlutterPointerMouseButtons::Primary,
-                            );
+                        WindowEvent::CursorEntered { device_id, .. } => {
+                            let idx = device_id.get_index();
+                            if active_devices.contains(idx) {
+                                eprintln!("{:?} already exists", device);
+                            }
+                            else {
+                                active_devices.insert(idx);
```
Comment (dvc94ch) [inline flutter-winit/src/window.rs]:
```rust
let id = if let Some(id) = active_devices.iter().position(|id| id == device_id) {
    id
} else {
    let id = active_devices.len();
    active_devices.push(device_id);
    id
}
```

**Comment by dvc94ch** [comment]
Superseded by #113

**Pull request closed without merging**